quint/src/quintParserFrontend.ts

```diff
@@ -128,6 +129,101 @@ export function parsePhase1(
   }
 }
 
+/**
+ * Phase 1b of the Quint parser. Go over each definition of the form
+ * `import ... from '<path>'`, do the following:
+ * 
+ *  - parse the modules that are refferred by each path,
+ *  - add the parsed modules.
+ * 
+ * Cyclic dependencies among different files are reported as errors.
+ */
+export function parsePhase1b(
```

Comment:
I got a sense of what this is doing, but how/why it works is not completely clear to me. It's 85 lines of imperative code, and I find it hard to follow. I'd much rather see this broken down into functions so I can reason about cycle detection, sorting and recursion separately.

I don't want to block this since we are just trying to get this working to unblock other work, but I do want to note the problem.

Reply:
Yeah, it's not always easy to express the same thing in functional code. This code is using the [Worklist algorithm](https://en.wikipedia.org/wiki/Reaching_definition#Worklist_algorithm). The challenge here is to also detect which modules should go first, hence, it assigns ranks to modules. I think I could rewrite it into functional code, but it would be even less obvious than the imperative code.

Reply:
If you've referenced that algo in your implementation, would help to add a comment citing that.